music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,552 @@
+        # Note: blank analyst and proof reader entries until supported within music21 metadata
+
+        if not self.container.getElementsByClass('TimeSignature'):
+            self.container.insert(0, meter.TimeSignature('4/4'))  # Placeholder
+
+        self.currentKeyString: str = ''
+        self.prepSequentialListOfLines()
+
+    def _makeContainer(self,
+                       obj: Union[stream.Stream, list]):
+        '''
+        Makes a placeholder container for the unusual cases where this class is called on
+        generic- or non-stream object as opposed to
+        a :class:`~music21.stream.Score`, :class:`~music21.stream.Part`,
+        or :class:`~music21.stream.Measure`.
+        '''
+        m = stream.Measure()
+        for x in obj:
+            m.append(x)
+        self.container = stream.Part()
+        self.container.insert(0, m)
+
+    def prepTitle(self,
+                  md: metadata):
+        '''
+        Attempt to prepare a single work title from the score metadata looking at each of
+        the title, movementNumber and movementName attributes.
+        Failing that, a placeholder 'Title unknown' stands in.
+
+        >>> s = stream.Score()
+        >>> rnScore = romanText.writeRoman.RnWriter(s)
+        >>> rnScore.title
+        'Title unknown'
+
+        >>> s.insert(0, metadata.Metadata())
+        >>> s.metadata.title = 'Fake title'
+        >>> s.metadata.movementNumber = 123456789
+        >>> s.metadata.movementName = 'Fake movementName'
+        >>> rnScoreWithMD = romanText.writeRoman.RnWriter(s)
+        >>> rnScoreWithMD.title
+        'Fake title - No.123456789: Fake movementName'
+        '''
+
+        workingTitle = []
+
+        if md.title:
+            workingTitle.append(md.title)
+        if md.movementNumber:
+            workingTitle.append(f'- No.{md.movementNumber}:')  # Spaces later
+        if md.movementName:
+            if md.movementName != md.title:
+                workingTitle.append(md.movementName)
+
+        if len(workingTitle) > 0:
+            self.title = ' '.join(workingTitle)
+
+# ------------------------------------------------------------------------------
+
+    def prepSequentialListOfLines(self):
+        '''
+        Prepares a sequential list of text lines, with time signatures and Roman numerals
+        adding this to the (already prepared) metadata preamble ready for printing.
+
+        >>> p = stream.Part()
+        >>> m = stream.Measure()
+        >>> m.insert(0, meter.TimeSignature('4/4'))
+        >>> m.insert(0, roman.RomanNumeral('V', 'G'))
+        >>> p.insert(0, m)
+        >>> testCase = romanText.writeRoman.RnWriter(p)
+        >>> testCase.combinedList[-1]  # Last entry, after the metadata
+        'm0 b1 G: V'
+        '''
+
+        for thisMeasure in self.container.getElementsByClass('Measure'):
+            # TimeSignatures
+            tsThisMeasure = thisMeasure.getElementsByClass('TimeSignature')
```

Comment:
Key Signatures are one of the things missing here -- at least make a TODO for it here.  (But not a block)